streaming/stream_session.cc

```diff
@@ -193,20 +193,34 @@ void stream_manager::init_messaging_service_handler(abort_source& as) {
                     }
                 });
             };
+          using received_partitions_t = std::pair<uint64_t, std::unique_ptr<dht::auto_refreshing_sharder>>;
           auto sink = _ms.local().make_sink_for_stream_mutation_fragments(source);
           try {
-            // Make sure the table with cf_id is still present at this point.
-            // Close the sink in case the table is dropped.
+           auto distribute = [this, cf_id, s] (flat_mutation_reader_v2 producer,
```

Comment:
You can mention in the commit log. This is trying to handle the case the sender does not know the table is dropped while the receiver thinks the table is dropped because schema propagation is async.